pkgs/development/python-modules/triton/default.nix

```diff
@@ -69,20 +69,40 @@ buildPythonPackage {
       })
     ];
 
-  postPatch = ''
+  postPatch =
     # Use our `cmakeFlags` instead and avoid downloading dependencies
     # remove any downloads
-    substituteInPlace python/setup.py \
-      --replace-fail "get_json_package_info(), get_pybind11_package_info()" ""\
-      --replace-fail "get_pybind11_package_info(), get_llvm_package_info()" ""\
-      --replace-fail 'packages += ["triton/profiler"]' ""\
-      --replace-fail "curr_version != version" "False"
-
+    ''
+      substituteInPlace python/setup.py \
+        --replace-fail "get_json_package_info(), get_pybind11_package_info()" ""\
+        --replace-fail "get_pybind11_package_info(), get_llvm_package_info()" ""\
+        --replace-fail 'packages += ["triton/profiler"]' ""\
+        --replace-fail "curr_version != version" "False"
+    ''
     # Don't fetch googletest
-    substituteInPlace unittest/CMakeLists.txt \
-      --replace-fail "include (\''${CMAKE_CURRENT_SOURCE_DIR}/googletest.cmake)" ""\
-      --replace-fail "include(GoogleTest)" "find_package(GTest REQUIRED)"
-  '';
+    + ''
+      substituteInPlace unittest/CMakeLists.txt \
+        --replace-fail "include (\''${CMAKE_CURRENT_SOURCE_DIR}/googletest.cmake)" ""\
+        --replace-fail "include(GoogleTest)" "find_package(GTest REQUIRED)"
+    ''
+    # Patch the triton source to not use ldconfig.
+    # https://github.com/triton-lang/triton/blob/2939d86fc5c4bbb64fd04fd5346a6dbed3bc3c85/third_party/nvidia/backend/driver.py#L27
+    + ''
+      substituteInPlace "$NIX_BUILD_TOP/$sourceRoot/third_party/nvidia/backend/driver.py" \
+        --replace-fail \
+          'libs = subprocess.check_output(["/sbin/ldconfig", "-p"]).decode()' \
+          'libs = ""'
```

Comment:
This was previously handled by `pkgs/development/python-modules/triton/0002-nvidia-amd-driver-short-circuit-before-ldconfig.patch`, you running into this error again means that patch is insufficient now. I think we should update the patch instead of extending `postPatch`

Reply:
My bad, it was failing because in my version of `cuda-packages` the `stubs` directory exists only in the `stubs` output; I was able to get it working without this change by getting the stubs output of `cuda_cudart`.

Reply:
Aah... well, we did merge `getFirstOutput [ "stubs" "lib" ]`

Reply:
Wait. This thread was about the patch, not about the stubs part. And the patch is not in the diff, did it get lost in the push?

Reply:
It's unnecessary because I changed the value we're templating the existing patch with; instead of using `cuda_cudart.lib` it now uses `cuda_cudart.stubs` because we want the stubs, specifically.

Reply:
Huh? I think the function in `driver.py` is used for detecting the real driver?

Reply:
Or, well, they probably reuse it for both purposes, the linkage and the dynamic loading...

Reply:
Staring at this line from PR description and trying to match it with the diff:

> finding libcuda.so will fall back to LD_LIBRARY_PATH per https://github.com/triton-lang/triton/commit/0149bf70042efb998fc7a1ea30ed99e3a0f75053

Side note: LD_LIBRARY_PATH for sure is about the real driver

Reply:
Good catch! I updated the description to instead say

```
- template `./0002-nvidia-amd-driver-short-circuit-before-ldconfig.patch` with path to CUDA stubs in `stubs` output rather than stub symlinks in `lib` output
```

Reply:
Ah, OK, I think I got it.

> My bad, it was failing because in my version of cuda-packages the stubs directory exists only in the stubs output; I was able to get it working without this change by getting the stubs output of cuda_cudart.

So the libcuda-related failure in `torch.compile` wasn't occurring with upstream cudaPackages, but with your next-gen cuda-packages?

And then the ptxas error was relevant to both?

Reply:
That's correct; sorry I wasn't clear about that to start with!